ops: align versions in monorepo

Kavinjsir opened the pull request:
https://github.com/rilldata/rill/issues/5592

himadrisingh commented:
lets make them `0.51.0` as the next release would be cut from this branch.

ericpgreen2 reviewed (commented):
Given 1) we're not publishing any of our packages to npm, 2) we have our own way of managing code versions via git tags, and 3) we're not consuming any `version` anywhere.... what if we removed the `version` fields altogether?

Rather:
- We could add `"private": true` to our package.json files to be more explicit that they're not published to `npm`.
- We could cite `web-common` in our `devDependencies` like: `"web-common": "workspace:*"`, which seems to be a standard practice in monorepos.

Thoughts?

Kavinjsir commented:
> Given 1) we're not publishing any of our packages to npm, 2) we have our own way of managing code versions via git tags, and 3) we're not consuming any `version` anywhere.... what if we removed the `version` fields altogether?
> 
> Rather:
> 
>     * We could add `"private": true` to our package.json files to be more explicit that they're not published to `npm`.
> 
>     * We could cite `web-common` in our `devDependencies` like: `"web-common": "workspace:*"`, which seems to be a standard practice in monorepos.
> 
> 
> Thoughts?

@ericpgreen2 That sounds great to me! It helps reduce the complexity of version maintenance across different workspaces and during the release process. I also tried update `version: "0.0.0"` for each workspace in the project, in an attempt to explicitly "lock" the version value in package.json.
One issue here is, when specifying `"web-common": "workspace:*"`, it would crash by the following error:
<img width="745" alt="workspace-issue" src="https://github.com/user-attachments/assets/e47787c6-99c3-49f1-bb2c-644c09762fec">

Not quite an expert over here, so I tried replace `"workspace:*"` with `"*"` to resolve the issue.
Here is the reference I got: https://stackoverflow.com/a/79135939

How would you like this?

cc: @himadrisingh I'm following @ericpgreen2 's suggestion which looks a good approach to reduce complexity. I also updated the version in the entry package.json to `0.51.0` regarding your suggestion :)

ericpgreen2 commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@rilldata/dev",
-  "version": "0.10.0",
+  "version": "0.51.0",
```
My comments on not relying on any `version` also apply to this root package.json. What if we set this one to `"0.0.0"` too?

Kavinjsir replied:
Sounds good!

ericpgreen2 commented on web-auth/package.json:
```diff
@@ -1,6 +1,7 @@
 {
 	"name": "web-auth",
-	"version": "0.0.1",
+  "private": true,
```
Redundant `private: true` (see line 5)

Kavinjsir replied:
Good Catch!

ericpgreen2 reviewed (approved):
LGTM! This seems like a good set-it-and-forget it approach for now. If we ever change how we version our releases, or if we ever publish some packages to npm, then we can revisit.
